### ISSUE TITLE
I added .NET Aspire and it stuck

### ISSUE THREAD

**Issue by tomokusaba** [microsoft/fluentui-blazor#987]
# 🐛 Bug Report

I added .NET Aspire and it stuck
## 💻 Repro or Code Sample
[before]
https://github.com/tomokusaba/BlazorApp27/tree/FluentUIBlazor

[after]
https://github.com/tomokusaba/BlazorApp27/tree/Aspire

## 🤔 Expected Behavior
Make it work even if you add .NET Aspire

## 😯 Current Behavior
![image](https://github.com/microsoft/fluentui-blazor/assets/46523924/0734dea6-894e-493a-a1ec-81f12e057025)
```
Unhandled exception. System.AggregateException: One or more errors occurred. (Method not found: 'Void Microsoft.FluentUI.AspNetCore.Components.LibraryConfiguration.set_HostingModel(Microsoft.FluentUI.AspNetCore.Components.BlazorHostingModel)'.)
 ---> System.MissingMethodException: Method not found: 'Void Microsoft.FluentUI.AspNetCore.Components.LibraryConfiguration.set_HostingModel(Microsoft.FluentUI.AspNetCore.Components.BlazorHostingModel)'.
   at Aspire.Dashboard.DashboardWebApplication.<>c.<.ctor>b__7_2(LibraryConfiguration options)
   at Microsoft.FluentUI.AspNetCore.Components.ServiceCollectionExtensions.AddFluentUIComponents(IServiceCollection services, Action`1 configuration) in /_/src/Core/Extensions/ServiceCollectionExtensions.cs:line 41
   at Aspire.Dashboard.DashboardWebApplication..ctor(ILogger`1 logger, Action`1 configureServices) in /_/src/Aspire.Dashboard/DashboardWebApplication.cs:line 93
   at Aspire.Hosting.Dcp.DcpHostService..ctor(DistributedApplicationModel applicationModel, DistributedApplicationOptions options, ILoggerFactory loggerFactory, IOptions`1 dcpOptions, IOptions`1 publishingOptions, ApplicationExecutor appExecutor, Locations locations, KubernetesService kubernetesService) in /_/src/Aspire.Hosting/Dcp/DcpHostService.cs:line 60
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
   at System.Reflection.MethodBaseInvoker.InvokeWithManyArgs(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at System.Reflection.RuntimeConstructorInfo.Invoke(BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitConstructor(ConstructorCallSite constructorCallSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSiteMain(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitRootCache(ServiceCallSite callSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSite(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitIEnumerable(IEnumerableCallSite enumerableCallSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSiteMain(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitRootCache(ServiceCallSite callSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSite(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.Resolve(ServiceCallSite callSite, ServiceProviderEngineScope scope)
   at Microsoft.Extensions.DependencyInjection.ServiceProvider.CreateServiceAccessor(ServiceIdentifier serviceIdentifier)
   at System.Collections.Concurrent.ConcurrentDictionary`2.GetOrAdd(TKey key, Func`2 valueFactory)
   at Microsoft.Extensions.DependencyInjection.ServiceProvider.GetService(ServiceIdentifier serviceIdentifier, ServiceProviderEngineScope serviceProviderEngineScope)
   at Microsoft.Extensions.DependencyInjection.ServiceProvider.GetService(Type serviceType)
   at Microsoft.Extensions.DependencyInjection.ServiceProviderServiceExtensions.GetRequiredService(IServiceProvider provider, Type serviceType)
   at Microsoft.Extensions.DependencyInjection.ServiceProviderServiceExtensions.GetRequiredService[T](IServiceProvider provider)
   at Microsoft.Extensions.Hosting.Internal.Host.StartAsync(CancellationToken cancellationToken)
   at Microsoft.Extensions.Hosting.HostingAbstractionsHostExtensions.RunAsync(IHost host, CancellationToken token)
   at Microsoft.Extensions.Hosting.HostingAbstractionsHostExtensions.RunAsync(IHost host, CancellationToken token)
   at Aspire.Hosting.DistributedApplication.RunAsync(CancellationToken cancellationToken) in /_/src/Aspire.Hosting/DistributedApplication.cs:line 139
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at System.Threading.Tasks.Task.Wait()
   at Aspire.Hosting.DistributedApplication.Run() in /_/src/Aspire.Hosting/DistributedApplication.cs:line 147
   at Program.<Main>$(String[] args) in C:\Users\tomo_\source\repos\BlazorApp27\BlazorApp27.AppHost\Program.cs:line 5
```

## 🌍 Your Environment

* .NET 8
* Fluent UI Blazor 4.0.0
* Aspire 8.0.0-preview.1.23557.2

**Comment by vnbaaij** [comment]
Aspire is still on our library version RC1. Go back to that version for your project as well or wait until Aspire is updated (believe end of November /begin December)

**Comment by AClerbois** [comment]
Hi @vnbaaij, 

I've got the same issue when I tried to use Aspire with a solution based on your latest version. 

I analyzed the issue and found the change eac3021 (#917). I suspect your reviewer is lazy (@dvoituron) 😄 . 

You forget to remove the ```BlazorHostingModel.cs``` (path : src\Core\Enums) file. This file seems no longer used. 

In addition, in your upgrade docs, you don't indicate the need to remove the setter of HostingModel in the ```AddFluentUIComponents``` IService extension. (file : examples\Demo\Shared\wwwroot\docs\UpgradeGuide.md). 

I propose to add a comment in your upgrade file to help people like me search about this issue.

With love to your awesome project. 

Br,

Adrien
